### ISSUE TITLE
JSON: attributes are not distinguished from strings

### ISSUE THREAD

**Issue by csabahenk** [sharkdp/bat#1400]
**What version of `bat` are you using?**

0.17.1

**Describe the bug you encountered:**

The highlighted output for JSON looks like this:

![image](https://user-images.githubusercontent.com/43937/100288511-226b4080-2f77-11eb-9285-b245ad1609ac.png)

according to the https://github.com/sharkdp/bat/blob/v0.17.1/tests/syntax-tests/highlighted/JSON/test.json test data file.

I haven't run the test suite, but verified that the executable produces the same result:

```
~/rust/src/bat $ /tmp/bat-v0.17.1-x86_64-unknown-linux-gnu/bat tests/syntax-tests/source/JSON/test.json -p --color=always | md5sum - tests/syntax-tests/highlighted/JSON/test.json

7351ba203005a061be42e64995b9982c  -
7351ba203005a061be42e64995b9982c  tests/syntax-tests/highlighted/JSON/test.json
```

**What did you expect to happen instead?**

Something like we get from an alternative highlighter on the source of the test (https://github.com/sharkdp/bat/blob/v0.17.1/tests/syntax-tests/source/JSON/test.json):

![image](https://user-images.githubusercontent.com/43937/100289002-37949f00-2f78-11eb-8559-1b1326648e66.png)

Ie. attributes (keys) are distinguished from strings (scalar values). As these two are semantically different, I expect them to look different.

**How did you install `bat`?**

Using the binary from Github release v0.17.1

**Comment by csabahenk** [comment]
Tthis might be deemed to be a duplicate of #1099; I filed a new issue because
- wanted to be more specific about my gripe with JSON than "doesn't look right"
- https://github.com/sharkdp/bat/issues/1099#issuecomment-667918309 seems to indicate that this issue is tacitly abandoded, even if kept open.

**Comment by keith-hall** [comment]
Thanks for reporting. It may be related to https://github.com/sublimehq/Packages/pull/862, where some scopes were changed - maybe most color schemes aren't targeting these still?

**Comment by sharkdp** [comment]
How can we move forward with this? This has been requested a few times and I also think it would be great if keys would be highlighted differently from string values.

I guess we would have to manually review the existing themes and propose upstream changes / patch the themes locally?

**Comment by keith-hall** [comment]
> I guess we would have to manually review the existing themes and propose upstream changes / patch the themes locally?

Yep. As an experiment, I tried adding the following to `Monokai Extended.tmTheme`:

```xml
    <dict>
      <key>name</key>
      <string>JSON String</string>
      <key>scope</key>
      <string>meta.mapping.key.json string.quoted.double.json</string>
      <key>settings</key>
      <dict>
        <key>foreground</key>
        <string>#66d9ef</string>
      </dict>
    </dict>
```

And here are the results:
![image](https://user-images.githubusercontent.com/11882719/103141998-d3473580-4705-11eb-87d4-a8dec96f72fb.png)

Ofc, I don't know what the most agreeable color to use would be, but it proves the theory at least :)

**Comment by sharkdp** [comment]
Thanks for looking into this, @keith-hall!

I think we should (at least) fix this for the default theme.

> EDIT: also: Merry Christmas! :)

Merry Christmas!

**Comment by sharkdp** [comment]
Somebody interested in fixing this by either providing a patch for the `Monokai Extended.tmTheme` theme in `assets/patches` or by forking the upstream repository with a patch?

**Comment by keith-hall** [comment]
> Somebody interested in fixing this by either providing a patch for the `Monokai Extended.tmTheme` theme in `assets/patches` or by forking the upstream repository with a patch?

I created a patch in https://github.com/sharkdp/bat/pull/1557

**Comment by sharkdp** [comment]
Fixed in #1557

**Comment by sharkdp** [comment]
Fixed in [bat v0.18](https://github.com/sharkdp/bat/releases/tag/v0.18.0).
